src/Symfony/Component/HttpKernel/EventListener/AbstractSessionListener.php

```diff
@@ -115,6 +116,34 @@ public function onKernelResponse(FilterResponseEvent $event)
              * it is saved will just restart it.
              */
             $session->save();
+
+            /*
+             * Write cookie to request object for cli context based runner like roadrunner,
+             * where php can not write the cookie.
+             */
+            if (\PHP_SAPI === 'cli') {
+                $cookieParams = session_get_cookie_params();
```

Comment:
Maybe we should inject this instead of build on top that the storage is setting this here per `ini_set`: https://github.com/symfony/symfony/blob/744b901750c1cc09e85bcb2ebdf0505449a1158f/src/Symfony/Component/HttpFoundation/Session/Storage/NativeSessionStorage.php#L395

Reply:
that'd be worth a try yes, instead of relying on global state